### PR TITLE
drag: bound piece within document

### PR DESCRIPTION
**Pull request by Zolmeister**
fixes https://github.com/ornicar/lila/pull/5276

### CODE REVIEW

File: src/drag.ts [hunk 1]
```diff
@@ -165,6 +165,12 @@ function processDrag(s: State): void {
         const translation = util.posToTranslateAbs(bounds)(cur.origPos, asWhite);
         translation[0] += cur.pos[0] + cur.dec[0];
         translation[1] += cur.pos[1] + cur.dec[1];
+        // bound piece within screen
+        const pieceBounds = cur.element.getBoundingClientRect()
```
Comment (isaacl) [inline src/drag.ts]: this is a scary function to call, especially in an animation tight loop.

**Comment by Zolmeister** [comment]
updated, removing `getBoundingClientRect()` call

File: src/drag.ts [hunk 2]
```diff
@@ -165,6 +168,11 @@ function processDrag(s: State): void {
         const translation = util.posToTranslateAbs(bounds)(cur.origPos, asWhite);
         translation[0] += cur.pos[0] + cur.dec[0];
         translation[1] += cur.pos[1] + cur.dec[1];
+        // bound piece within screen
+        const maxX = document.body.clientWidth + document.body.scrollLeft - cur.size[0] - 8.5;
+        const maxY = document.body.clientHeight + document.body.scrollTop - cur.size[1] - 8.5;
```
Comment (ornicar) [inline src/drag.ts]: What is this `8.5` value?

Comment (Zolmeister) [reply src/drag.ts]: I think it's caused by https://github.com/ornicar/chessground/blob/a722f739a9394b992f233907c6912d24d936502a/src/svg.ts#L251

Comment (ornicar) [reply src/drag.ts]: But it only makes sense when multiplying a width, as to reach a center.

**Comment by ornicar** [comment]
This is hackish and I don't want to call `document.body.clientWidth` or `document.body.scrollLeft` or any such measurements during piece animation. I like the current solution better.

**Pull request closed without merging**